inbm/dispatcher-agent/dispatcher/dispatcher_class.py

```diff
@@ -734,10 +734,10 @@ def handle_updates(dispatcher: Any,
         # TODO: Change single and repeated to add the schedules to the scheduler DB
         if schedule.single_scheduled_requests or schedule.repeated_scheduled_requests:
             sql_lock.acquire()
-            for single_task in schedule.single_scheduled_requests:
-                SqliteManager().create_task(single_task)
-            for repeated_task in schedule.repeated_scheduled_requests:
-                SqliteManager().create_task(repeated_task)
+            for ss in schedule.single_scheduled_requests:
+                SqliteManager().create_schedule(ss)
+            for rs in schedule.repeated_scheduled_requests:
+                SqliteManager().create_schedule(rs)
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** To avoid potential resource leaks and ensure the database connection is properly managed, it's recommended to instantiate SqliteManager once and reuse the instance for multiple operations. This approach minimizes the overhead of repeatedly creating and destroying objects, leading to better performance and resource utilization.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
            sqlite_manager = SqliteManager()
            for ss in schedule.single_scheduled_requests:
                sqlite_manager.create_schedule(ss)
            for rs in schedule.repeated_scheduled_requests:
                sqlite_manager.create_schedule(rs)
```